### PR TITLE
redpanda: fix linux build

### LINKED ISSUE #172784: Cannot Install redpanda
### Steps To Reproduce
1. `sudo nix-channel --list`
```zsh
    nixos https://channels.nixos.org/nixos-21.11
    nixos-unstable https://channels.nixos.org/nixos-unstable
```
2. `nix-env -iA nixos-unstable.redpanda`

### Build log
```zsh
installing 'redpanda-21.11.15'
these derivations will be built:
  /nix/store/kc6npr76yslbppm2qi2l1ny8xlxv5xbx-source.drv
  /nix/store/gxl7pmalwkkd6rrb9gqwspyxjs5fiqfb-redpanda-21.11.15.drv
these paths will be fetched (73.99 MiB download, 337.58 MiB unpacked):
  /nix/store/0fnnx443l6107v99719vd9n04wrznmas-xz-5.2.5-bin
  /nix/store/0nh5xi56ar68jj88amj91yckm5nr237w-nghttp2-1.47.0
  /nix/store/0smvqxwkwhrfd92fbim86kx3q41il0iv-unzip-6.0
  /nix/store/13y9c8f2jxwmszbnlp86qrzb1bvyl95a-gzip-1.12
  /nix/store/1wbkdhc44n09f6mhdw7y0d8dlkck29m9-gcc-11.2.0
  /nix/store/2d9wf0l8r5rn3wgbm25v5wpp9apfiq18-nghttp2-1.47.0-lib
  /nix/store/2lrlxvyr6sbq4i0yvsqrkcnrani8zw08-linux-headers-5.17
  /nix/store/2pg21nd0kzxfpahsb5j1awgd04g840j5-xz-5.2.5
  /nix/store/3n28q42pi8bwh8ry90n101xj92sr43km-nghttp2-1.47.0-bin
  /nix/store/4q49m81fzcwycnfspwbsc1iqk12qxpyq-patchelf-0.14.5
  /nix/store/4x0gih1klxpzf4ass44h7w937wljv9r7-stdenv-linux
  /nix/store/5158ikyyda1y8z707jg8spac49i8mxa3-pcre-8.45
  /nix/store/69brclzxp7mg927k6986hrfzyd1hpqgd-gcc-11.2.0-lib
  /nix/store/6ka6x238gwc0dxwvv32g8xl9l4wbnkqj-bzip2-1.0.6.0.2-bin
  /nix/store/7dc5r0a6xy7mglbi57hmhvcwm1nqjwiz-gnutar-1.34
  /nix/store/81fagzrg61w8hiy02bjrk9gvmrm6xqp6-curl-7.82.0-bin
  /nix/store/8v25w84b16ysam9nf6chflkyz1xhgziz-perl-5.34.1
  /nix/store/94rpgjbiayhb43vd3akm0fl8swgswqhq-glibc-2.34-115-bin
  /nix/store/af7l2h86h4b7j4379yc1yqi6dlxzfx8h-curl-7.82.0-dev
  /nix/store/apfkjcwpjj079alxv8kgx1ncb42jlks9-openssl-1.1.1n
  /nix/store/bd9k3zwm0b3p8k3qw0nlk71vzfa5hn8n-glibc-2.34-115-dev
  /nix/store/bzg7izn1a7sk30wjb1brq9kd7vvf07k1-libssh2-1.10.0
  /nix/store/c5ak1rgjn5l4ip9q7ggc3c01nzsm0yby-libkrb5-1.19.2
  /nix/store/ck51398a080fkq9j888iz8nwf2id54wd-expand-response-params
  /nix/store/cms5zic4spc5bvbbvqqpk94js1hh7dzh-keyutils-1.6.3-lib
  /nix/store/crp15hh3h207qpppnnpk0cmlnzcapp5x-curl-7.82.0-man
  /nix/store/f5vc4i92d0412g1bl3cjqb4d5pk661k6-coreutils-9.0
  /nix/store/g1jc160jp4063xq2y1ll939nw6l8albb-findutils-4.9.0
  /nix/store/hfiina0dmb8k0nfq8jz2yjccp9isjf1k-ed-1.18
  /nix/store/ik4qlj53grwmg7avzrfrn34bjf6a30ch-libunistring-1.0
  /nix/store/k0np19xv7wjc1grhrb2y6r85qiv1cikd-gcc-wrapper-11.2.0
  /nix/store/kakj7qy4l6spsds7fkfgz595hazn2rry-libkrb5-1.19.2-dev
  /nix/store/kfx4l9lq4cqrah8fb2mpr35xpcmk5gdx-mirrors-list
  /nix/store/kn3pmx1iyp7rgi6xvgclk7zldy05q7qk-openssl-1.1.1n-bin
  /nix/store/knii1xbbipchia5jfhwb12028028rb1y-zlib-1.2.12-dev
  /nix/store/lq8m46y7wdzx52jlzs6hz1pj5jwi3zk8-curl-7.82.0
  /nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115
  /nix/store/myrkvalb2zkvqnllbm6q9133jn4za1n9-bzip2-1.0.6.0.2
  /nix/store/n7p9j0nfywa85cisl7c1l7wzb8n7s9rq-binutils-wrapper-2.38
  /nix/store/phfby2kj0vk1c8a5sc58i2zakdz2l771-gnused-4.8
  /nix/store/pi4giz86cn828bd2hlv3sbdxg4k55531-gnumake-4.3
  /nix/store/qbz72c5q201mlyfib2010xjgdzbpiw74-stdenv-linux
  /nix/store/ra6q6ccjyp7hk3zphv69kr7nqqklggk0-nghttp2-1.47.0-dev
  /nix/store/rrpryjxr6f15bzdnyyp4n4pmymh145vr-openssl-1.1.1n-dev
  /nix/store/s4q2kylscmz7v5znpbz95rjkai0m16y8-gawk-5.1.1
  /nix/store/s7r5cnjj4zy41g2nql3vf2ivr09wfm5p-bash-5.1-p16
  /nix/store/sjcjlpdnj70l9p6fi0r0camnn6slah96-libssh2-1.10.0-dev
  /nix/store/vrail3wxl4i3vh288p2c0z01gs63fvrq-patch-2.7.6
  /nix/store/vybkpg9gqy10d8pm17wz6x1ljrcsfn5d-zlib-1.2.12
  /nix/store/vzsqcldzq3y8cwj0mr8pbc6x9w1jmfj1-binutils-2.38
  /nix/store/w3zngkrag7vnm7v1q8vnqb71q6a1w8gn-libidn2-2.3.2
  /nix/store/x3zm93mdhykfp51qjjcd8wlg3zw4v8jk-attr-2.5.1
  /nix/store/yjq4by8a5rp4ggll3082mrb1fmpai9cl-diffutils-3.8
  /nix/store/ykyw3g57pc61b24cn06kgyngsq4azh7y-gnugrep-3.7
  /nix/store/z7cy14zczhk4sslf9c79z25ivbci75bv-acl-2.3.1
copying path '/nix/store/2lrlxvyr6sbq4i0yvsqrkcnrani8zw08-linux-headers-5.17' from 'https://cache.nixos.org'...
copying path '/nix/store/ik4qlj53grwmg7avzrfrn34bjf6a30ch-libunistring-1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/kfx4l9lq4cqrah8fb2mpr35xpcmk5gdx-mirrors-list' from 'https://cache.nixos.org'...
copying path '/nix/store/crp15hh3h207qpppnnpk0cmlnzcapp5x-curl-7.82.0-man' from 'https://cache.nixos.org'...
copying path '/nix/store/0nh5xi56ar68jj88amj91yckm5nr237w-nghttp2-1.47.0' from 'https://cache.nixos.org'...
copying path '/nix/store/3n28q42pi8bwh8ry90n101xj92sr43km-nghttp2-1.47.0-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/w3zngkrag7vnm7v1q8vnqb71q6a1w8gn-libidn2-2.3.2' from 'https://cache.nixos.org'...
copying path '/nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115' from 'https://cache.nixos.org'...
copying path '/nix/store/s7r5cnjj4zy41g2nql3vf2ivr09wfm5p-bash-5.1-p16' from 'https://cache.nixos.org'...
copying path '/nix/store/hfiina0dmb8k0nfq8jz2yjccp9isjf1k-ed-1.18' from 'https://cache.nixos.org'...
copying path '/nix/store/x3zm93mdhykfp51qjjcd8wlg3zw4v8jk-attr-2.5.1' from 'https://cache.nixos.org'...
copying path '/nix/store/s4q2kylscmz7v5znpbz95rjkai0m16y8-gawk-5.1.1' from 'https://cache.nixos.org'...
copying path '/nix/store/ck51398a080fkq9j888iz8nwf2id54wd-expand-response-params' from 'https://cache.nixos.org'...
copying path '/nix/store/myrkvalb2zkvqnllbm6q9133jn4za1n9-bzip2-1.0.6.0.2' from 'https://cache.nixos.org'...
copying path '/nix/store/69brclzxp7mg927k6986hrfzyd1hpqgd-gcc-11.2.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/94rpgjbiayhb43vd3akm0fl8swgswqhq-glibc-2.34-115-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/pi4giz86cn828bd2hlv3sbdxg4k55531-gnumake-4.3' from 'https://cache.nixos.org'...
copying path '/nix/store/phfby2kj0vk1c8a5sc58i2zakdz2l771-gnused-4.8' from 'https://cache.nixos.org'...
copying path '/nix/store/13y9c8f2jxwmszbnlp86qrzb1bvyl95a-gzip-1.12' from 'https://cache.nixos.org'...
copying path '/nix/store/cms5zic4spc5bvbbvqqpk94js1hh7dzh-keyutils-1.6.3-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/2d9wf0l8r5rn3wgbm25v5wpp9apfiq18-nghttp2-1.47.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/z7cy14zczhk4sslf9c79z25ivbci75bv-acl-2.3.1' from 'https://cache.nixos.org'...
copying path '/nix/store/6ka6x238gwc0dxwvv32g8xl9l4wbnkqj-bzip2-1.0.6.0.2-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/apfkjcwpjj079alxv8kgx1ncb42jlks9-openssl-1.1.1n' from 'https://cache.nixos.org'...
copying path '/nix/store/vrail3wxl4i3vh288p2c0z01gs63fvrq-patch-2.7.6' from 'https://cache.nixos.org'...
copying path '/nix/store/ra6q6ccjyp7hk3zphv69kr7nqqklggk0-nghttp2-1.47.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/c5ak1rgjn5l4ip9q7ggc3c01nzsm0yby-libkrb5-1.19.2' from 'https://cache.nixos.org'...
copying path '/nix/store/f5vc4i92d0412g1bl3cjqb4d5pk661k6-coreutils-9.0' from 'https://cache.nixos.org'...
copying path '/nix/store/7dc5r0a6xy7mglbi57hmhvcwm1nqjwiz-gnutar-1.34' from 'https://cache.nixos.org'...
copying path '/nix/store/5158ikyyda1y8z707jg8spac49i8mxa3-pcre-8.45' from 'https://cache.nixos.org'...
copying path '/nix/store/0smvqxwkwhrfd92fbim86kx3q41il0iv-unzip-6.0' from 'https://cache.nixos.org'...
copying path '/nix/store/bd9k3zwm0b3p8k3qw0nlk71vzfa5hn8n-glibc-2.34-115-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/4q49m81fzcwycnfspwbsc1iqk12qxpyq-patchelf-0.14.5' from 'https://cache.nixos.org'...
copying path '/nix/store/2pg21nd0kzxfpahsb5j1awgd04g840j5-xz-5.2.5' from 'https://cache.nixos.org'...
copying path '/nix/store/kakj7qy4l6spsds7fkfgz595hazn2rry-libkrb5-1.19.2-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/vybkpg9gqy10d8pm17wz6x1ljrcsfn5d-zlib-1.2.12' from 'https://cache.nixos.org'...
copying path '/nix/store/ykyw3g57pc61b24cn06kgyngsq4azh7y-gnugrep-3.7' from 'https://cache.nixos.org'...
copying path '/nix/store/g1jc160jp4063xq2y1ll939nw6l8albb-findutils-4.9.0' from 'https://cache.nixos.org'...
copying path '/nix/store/yjq4by8a5rp4ggll3082mrb1fmpai9cl-diffutils-3.8' from 'https://cache.nixos.org'...
copying path '/nix/store/0fnnx443l6107v99719vd9n04wrznmas-xz-5.2.5-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/1wbkdhc44n09f6mhdw7y0d8dlkck29m9-gcc-11.2.0' from 'https://cache.nixos.org'...
copying path '/nix/store/vzsqcldzq3y8cwj0mr8pbc6x9w1jmfj1-binutils-2.38' from 'https://cache.nixos.org'...
copying path '/nix/store/8v25w84b16ysam9nf6chflkyz1xhgziz-perl-5.34.1' from 'https://cache.nixos.org'...
copying path '/nix/store/bzg7izn1a7sk30wjb1brq9kd7vvf07k1-libssh2-1.10.0' from 'https://cache.nixos.org'...
copying path '/nix/store/knii1xbbipchia5jfhwb12028028rb1y-zlib-1.2.12-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/lq8m46y7wdzx52jlzs6hz1pj5jwi3zk8-curl-7.82.0' from 'https://cache.nixos.org'...
copying path '/nix/store/sjcjlpdnj70l9p6fi0r0camnn6slah96-libssh2-1.10.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/qbz72c5q201mlyfib2010xjgdzbpiw74-stdenv-linux' from 'https://cache.nixos.org'...
copying path '/nix/store/81fagzrg61w8hiy02bjrk9gvmrm6xqp6-curl-7.82.0-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/n7p9j0nfywa85cisl7c1l7wzb8n7s9rq-binutils-wrapper-2.38' from 'https://cache.nixos.org'...
copying path '/nix/store/kn3pmx1iyp7rgi6xvgclk7zldy05q7qk-openssl-1.1.1n-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/rrpryjxr6f15bzdnyyp4n4pmymh145vr-openssl-1.1.1n-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/af7l2h86h4b7j4379yc1yqi6dlxzfx8h-curl-7.82.0-dev' from 'https://cache.nixos.org'...
building '/nix/store/kc6npr76yslbppm2qi2l1ny8xlxv5xbx-source.drv'...
copying path '/nix/store/k0np19xv7wjc1grhrb2y6r85qiv1cikd-gcc-wrapper-11.2.0' from 'https://cache.nixos.org'...
copying path '/nix/store/4x0gih1klxpzf4ass44h7w937wljv9r7-stdenv-linux' from 'https://cache.nixos.org'...

trying https://github.com/redpanda-data/redpanda/releases/download/v21.11.15/rpk-linux-amd64.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 12.2M  100 12.2M    0     0  1703k      0  0:00:07  0:00:07 --:--:-- 2249k
unpacking source archive /build/rpk-linux-amd64.zip
hash mismatch in fixed-output derivation '/nix/store/y6s05qcli1q7czc84qjax3825as1k1cj-source':
  wanted: sha256:0ipmi3viw8ar7qgqbdh8s2ci4d9fwcg4y1acdzkfjzswapp0xzk1
  got:    sha256:0vpxn7kcpqylk0nc74m6yxgwwf8ns8pyb6kxnmnmv2x58f8x4c8n
cannot build derivation '/nix/store/gxl7pmalwkkd6rrb9gqwspyxjs5fiqfb-redpanda-21.11.15.drv': 1 dependencies couldn't be built
error: build of '/nix/store/gxl7pmalwkkd6rrb9gqwspyxjs5fiqfb-redpanda-21.11.15.drv' failed
```

### Notify maintainers

@happysalada 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.113, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11.337422.3c5ae9be1f1, nixos-unstable-22.05pre375061.c777cdf5c56"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by happysalada**
###### Description of changes

closes https://github.com/NixOS/nixpkgs/issues/172784
verifying the ci passes before merging.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/redpanda/default.nix [hunk 1]
```diff
@@ -4,13 +4,21 @@ let
   version = "21.11.15";
   platform = if stdenv.isLinux then "linux" else "darwin";
   arch = if stdenv.isAarch64 then "arm" else "amd";
+  sha256s = {
+    darwin.amd = "sha256-Yf4O7lVcf+nmb0wFTx7jLjUSmdAItoUfPlkhHveI9UY=";
+    # TODO if you have this aarch, feel free to submit a PR for the sha256
+    darwin.arm = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
```
Comment (bobby285271) [inline pkgs/servers/redpanda/default.nix:10]:
```suggestion
    darwin.arm = "sha256-vKfFBheDZtvcbg0zbj3rqUEQczZvySuGuM3RopnDJG8=";
```

File: pkgs/servers/redpanda/default.nix [hunk 2]
```diff
@@ -4,13 +4,21 @@ let
...
+    linux.amd = "sha256:0vpxn7kcpqylk0nc74m6yxgwwf8ns8pyb6kxnmnmv2x58f8x4c8n";
+    # TODO if you have this aarch, feel free to submit a PR for the sha256
+    linux.arm = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
```
Comment (bobby285271) [inline pkgs/servers/redpanda/default.nix:13]:
```suggestion
    linux.arm = "sha256-1AQSB2V5zGivU0UinTST2kOydQf/bmLbpjdW0Yo4ptE=";
```

**Review by bobby285271 — COMMENTED** [review]
Got these hash using `nix-prefetch fetchzip --url <url>`, verifed that the other 2 hashes are correct.

**Comment by happysalada** [comment]
@bobby285271 thank you!

**Pull request merged**